simbad/rotsearch/amore_search.py

```diff
@@ -378,169 +102,127 @@ def run_pdb(self, models_dir, mr_model_dir, nproc=2, shres=3.0, pklim=0.5, npic=
             log file for each model in the models_dir
 
         """
-        simbad_dat_files = [
-            os.path.join(root, filename) for root, _, files in os.walk(models_dir)
-            for filename in files if filename.endswith('.dat')
-        ]
+        simbad_dat_files = simbad.db.find_simbad_dat_files(models_dir)
 
-        tmp_dir = os.path.join(self.work_dir, 'tmp')
-        if not os.path.isdir(tmp_dir):
-            os.makedirs(tmp_dir)
+        space_group, _, cell = simbad.util.mtz_util.crystal_data(self.mtz)
+        cell = " ".join(map(str, cell))
 
-        space_group, _, cell_parameters = mtz_util.crystal_data(self.mtz)
-        cell_parameters = " ".join(map(str, cell_parameters))
+        chunk_size = AmoreRotationSearch.get_chunk_size(len(simbad_dat_files),
+                                                        chunk_size)
+        total_chunk_cycles = AmoreRotationSearch.get_total_chunk_cycles(len(simbad_dat_files),
+                                                                        chunk_size)
 
-        chunk_size = len(simbad_dat_files) if chunk_size == 0 else chunk_size
-        total_chunk_cycles, remainder = divmod(len(simbad_dat_files),
-                                               chunk_size)
-        if remainder > 0:
-            total_chunk_cycles += 1
-
-        rotation_data = []
-        for cycle, i in enumerate(range(0, len(simbad_dat_files), chunk_size)):
-            logger.info("Working on chunk %d out of %d",
-                        cycle + 1, total_chunk_cycles)
-            chunk_dat_files = simbad_dat_files[i:i + chunk_size]
-
-            # Generate the relevant scripts and data
-            amore_files, to_delete = [], []
-            for dat_model in chunk_dat_files:
-                root = dat_model.replace('.dat', '')
-                name = os.path.basename(root)
-
-                input_model = tmp_file(directory=tmp_dir, prefix="",
-                                       stem=name, suffix='.pdb')
-                with open(dat_model, 'rb') as f_in, open(input_model, 'w') as f_out:
-                    f_out.write(zlib.decompress(base64.b64decode(f_in.read())))
-
-                try:
-                    solvent_content = matthews_coef.solvent_content(input_model,
-                                                                    cell_parameters,
-                                                                    space_group)
-                except ValueError:
-                    logger.critical("Error calculating solvent content for %s",
-                                    name)
-                    continue
+        tmp_dir = os.path.join(self.work_dir, "tmp")
+        if not os.path.isdir(tmp_dir):
+            os.makedirs(tmp_dir)
 
+        hklpck0 = self._generate_hklpck0()
+	template_hklpck1 = os.path.join(tmp_dir, "{}.hkl")
```

Comment:
Can you double check the indexing here, it seems to be one space off

Reply:
Addressed in 35f4195